test/functional/TestUtilities/src/org/openj9/test/attachAPI/TargetManager.java

```diff
@@ -106,16 +109,29 @@ public TargetManager(String cmdName, String targetId) {
 
 	/**
 	 * Wait until the JVM's attach API has initialized
-	 * @return success if true, false if attach API is disabled or error
+	 * @return success if true, false if attach API is disabled or error, or does not initialize in 100 s.
 	 */
 	public static boolean waitForAttachApiInitialization() {
 		boolean result = false;
-		try {
-			Class<?> attachHandlerClass = Class.forName(TargetManager.COM_IBM_TOOLS_ATTACH_TARGET_ATTACH_HANDLER);
-			final Method waitForAttachApiInitialization = attachHandlerClass.getMethod("waitForAttachApiInitialization");
-			result = (boolean) waitForAttachApiInitialization.invoke(attachHandlerClass);
-		} catch (ClassNotFoundException | NoSuchMethodException | SecurityException | IllegalAccessException | IllegalArgumentException | InvocationTargetException e) {
-			logger.error("error waiting for attach API initialization: "+e.getMessage());
+		int tries = 100;
+		com.ibm.lang.management.RuntimeMXBean bean = (RuntimeMXBean) ManagementFactory.getRuntimeMXBean();
+		while (tries > 0) {
+			logger.debug("Poll attach API status");
+			if (bean.isAttachApiInitialized()) {
+				result = true;
+				logger.debug("attach API initialized");
+				break;
+			} else if (bean.isAttachApiTerminated()) {
+				logger.debug("attach API terminated");
+				result = false;
+				break;
+			} 
+			try {
+				Thread.sleep(1000);
```

Comment:
Seems like a long time to wait. Can this be reduced?

Reply:
Note that the delay is in milliseconds: if the attach API hasn't initialized by the time we hit this code, it will likely be several seconds before it is initialized.  The check usually succeeds on the first pass.  Also, subsequent calls will return immediately because the attach API will have settled into the initialized or terminated states.